coderd/files/cache.go

```diff
@@ -16,22 +18,75 @@ import (
 
 // NewFromStore returns a file cache that will fetch files from the provided
 // database.
-func NewFromStore(store database.Store) *Cache {
-	fetcher := func(ctx context.Context, fileID uuid.UUID) (fs.FS, error) {
+func NewFromStore(store database.Store, registerer prometheus.Registerer) *Cache {
+	fetch := func(ctx context.Context, fileID uuid.UUID) (fs.FS, int64, error) {
 		file, err := store.GetFileByID(ctx, fileID)
 		if err != nil {
-			return nil, xerrors.Errorf("failed to read file from database: %w", err)
+			return nil, 0, xerrors.Errorf("failed to read file from database: %w", err)
 		}
 
 		content := bytes.NewBuffer(file.Data)
-		return archivefs.FromTarReader(content), nil
+		return archivefs.FromTarReader(content), int64(content.Len()), nil
```

Comment:
hmm. I really dislike trusting honest reporting of size here. is there not some way we could inspect the size of this entire object from inside the cache at runtime? there might not be but it'd be super slick

Reply:
I don't think there is without doing some recursive `Stat()` call on all the files in the filesystem.

Right here is the cheapest place to do it. If we ever add a compression layer, then this won't be 100% accurate, but at present it does indicate the total number of bytes held in memory by the cache entry.